pkg/providers/tinkerbell/validator.go

```diff
@@ -283,3 +295,22 @@ func hardwareMap(hardwareList []*tinkhardware.Hardware) map[string]*tinkhardware
 
 	return hardwareMap
 }
+
+// workflowMap returns all the workflows on the tinkerbell stack in the form of map with mac address as a key
+func workflowMap(workflowList []*tinkworkflow.Workflow) (map[string]*tinkworkflow.Workflow, error) {
```

Comment:
nit: do the same for hardwareMap? Since both are returning a map it might add more clarity. 
```suggestion
func getWorkflowMap(workflowList []*tinkworkflow.Workflow) (map[string]*tinkworkflow.Workflow, error) {
```